pkg/relayer/protocol/difficulty.go

```diff
@@ -18,6 +18,17 @@ func MustCountDifficultyBits(bz []byte) int {
 
 // CountDifficultyBits returns the number of leading zero bits in the given byte
 // slice. It returns an error if the byte slice is all zero bits.
+//
+// TODO_IMPROVE(@h5law): Remove the forloop logic and replace with a simplified
+// single method that accounts for the fact that block hashes/paths are always
```

Comment:
I know I'm sounding like a broken record a bit but:

1. The assumption that block hahes will be 32 bytes is fair
2. The assumption that paths will remain 32 bytes I don't think should be stated. smt is a library so whether or not it will be configurable, we should tightly couple the two so much. 
3. We should change how we use hashes -> paths. The idea is that a `block hash` is a `seed` for the path and not the path itself. I was thinking `path = foo(block_hash, session_id)` and it does some sort of processing inside.

Reply:
> The assumption that paths will remain 32 bytes I don't think should be stated. smt is a library so whether or not it will be configurable, we should tightly couple the two so much.

I mean its because we are using `sha256` the `smt` library can have bigger or smaller paths but we have chosen a 32 byte hasher. I think this is something we should settle on and not change - maybe only change to https://pkg.go.dev/golang.org/x/crypto/sha3 `sha3.Sum256` I see no reason for this to be configurable as it would make things overly complex.

> We should change how we use hashes -> paths. The idea is that a block hash is a seed for the path and not the path itself. I was thinking path = foo(block_hash, session_id) and it does some sort of processing inside.

```go
seed := blockHash
seed = append(seed, []byte(serviceID))
path := sha256.Sum256(seed)
```

??? Does it have to be more complex than hashing some seed we can augment it with as much data as we want but hashing it needs to be done now as path sizes equalling one another is a requirement of the ClosestProof - unless this is not intended to be the same as the path provided to that?